src/tinyformat.h

```diff
@@ -705,23 +705,27 @@ inline const char* streamStateFromFormat(std::ostream& out, bool& spacePadPositi
             break;
         case 'X':
             out.setf(std::ios::uppercase);
+            [[fallthrough]];
```

Comment:
Upstream did this in https://github.com/bitcoin/bitcoin/pull/21430; please either backport that commit or check that you have everything relevant from it. (It appears to also have a change to the leveldb compilation flags.)